src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -539,21 +541,54 @@ end subroutine read_geogrid
 
         ierr = 0
 
+        tile_nx => null()
+        tile_ny => null()
+        tile_nz => null()
+        tile_z_start => null()
+        tile_z_end => null()
+        tile_bdr => null()
+        wordsize => null()
         tile => null()
         endian => null()
         signed => null()
 
+        err_level = mpas_pool_get_error_level()
+        call mpas_pool_set_error_level(MPAS_POOL_SILENT)
+
         call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
         call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
         call mpas_pool_get_config(mgr % pool, 'tile_z', tile_nz)
         call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
         call mpas_pool_get_config(mgr % pool, 'wordsize', wordsize)
         call mpas_pool_get_config(mgr % pool, 'signed', signed)
         call mpas_pool_get_config(mgr % pool, 'endian', endian)
+        call mpas_pool_get_config(mgr % pool, 'tile_z_start', tile_z_start)
+        call mpas_pool_get_config(mgr % pool, 'tile_z_end', tile_z_end)
+
+        ! Reset the pool's error level
+        call mpas_pool_set_error_level(err_level)
 
         c_tile_nx = tile_nx + 2 * tile_bdr ! The number of pixels in the x direction, including halo cells
         c_tile_ny = tile_ny + 2 * tile_bdr ! The number of pixels in the y direction, including halo cells
-        c_tile_nz = tile_nz ! The number of pixels in the z direction
+
+        !
+        ! Some datasets contain tile_z_start and tile_z_end, while some contain tile_z to describe
+        ! their third dimension. Although this generalization is not needed currently for any datasets
+        ! used by MPAS, we should ensure the bounds of the third dimension of tile % tile are allocated
+        ! to be between tile_z_start:tile_z_end incase a future dataset describes its z dimension as
+        ! anything other than 1:tile_z or 1:tile_z_end.
+        !
+        if (.not. associated(tile_nz)) then
+            allocate(tile_nz)
+            tile_nz = (tile_z_end - tile_z_start) + 1
+            c_tile_nz = tile_nz
+        else
+            allocate(tile_z_start)
+            allocate(tile_z_end)
```

Comment:
@mgduda I think you're correct that something not right is going on. Taking a second look at this this would always allocate `tile_z_start` and `tile_z_end` when `read_geogrid` is called... Perhaps I wrote this function before my first cup of coffee?

I'll move this logic up to the initialization function.